src/Compilers/CSharp/Portable/CSharpParseOptions.cs

```diff
@@ -36,6 +36,8 @@ public sealed class CSharpParseOptions : ParseOptions, IEquatable<CSharpParseOpt
 
         internal ImmutableArray<string> PreprocessorSymbols { get; private set; }
 
+        internal bool ThrowOnInsufficientExecutionStack { get; private set; }
```

Comment:
This is essentially a feature, essentially using threads to create chained stacks. Consider naming this around the feature, not the behavior for the lack of the feature. Example: `UseThreadsForDeeplyRecursiveStructures`. 

I've gone back and forth on this but wanted to call it out for others input.